docs/choreolib/auto-routines.md

```diff
@@ -1,7 +1,92 @@
 # Auto Routines
 
+## Setup
+
+### AutoFactory
+
 ChoreoLib provides the `AutoFactory` class as higher level API to make it easier to create competitive and complex auto routines inside your robot code.
 
+You can setup the `AutoFactory` by calling `Choreo.createAutoFactory`
+
+```java
+// The most basic usage of the AutoFactory
+class Robot extends TimedRobot {
+  /** A swerve drive subsystem */
+  private final Drive drive = ...;
+  /** An object that manages information about robot position */
+  private final Localizer localizer = ...;
+  private final AutoFactory autoFactory;
+
+  public Robot() {
+    autoFactory = Choreo.createAutoFactory(
+      drive,
+      localizer::pose,
+      (p, sample) -> drive.drive(new ChassisSpeeds(sample.vx, sample.vy, sample.omega)),
+      () -> DriverStation.getAlliance().orElseGet(() -> Alliance.Blue).equals(Alliance.Red),
+      new AutoBindings()
+    );
+  }
+
+  public void autonomousInit() {
+    // Running just the movement of a specific trajectory
+    autoFactory.trajectoryCommand("myTrajectory").schedule();
+  }
+}
+```
+
+### AutoChooser, AutoRoutine, and AutoTrajectory
+
+The `AutoFactory` can create `AutoRoutine` and `AutoTrajectory` objects that can be used to create complex auto routines.
+The `AutoChooser` provides a simple api to structure your auto routine generation in the most performant way.
```

Comment:
```suggestion
The `AutoChooser` provides a simple API to structure your auto routine constructors in the most performant way.
```

Reply:
I don't like constructors here, that word is already used in java

Reply:
factories is already used by `AutoFactory`... we're running out of words lmao, I think keeping it simple with just "creation" could be fine